pkg/utils/controller/policy_snapshot_resolver_test.go

```diff
@@ -680,6 +810,60 @@ func TestListPolicySnapshots(t *testing.T) {
 			},
 			expectedSnapshots: 2,
 		},
+		{
+			name: "list namespaced policy snapshots - mixed setup",
+			placementKey: types.NamespacedName{
+				Name:      placementName,
+				Namespace: policySnapshotNamespace,
+			},
+			existingSnapshots: []client.Object{
+				&fleetv1beta1.SchedulingPolicySnapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-placement-0",
+						Namespace: policySnapshotNamespace,
+						Labels: map[string]string{
+							fleetv1beta1.PlacementTrackingLabel: placementName,
+						},
+					},
+				},
+				&fleetv1beta1.SchedulingPolicySnapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-placement-1",
+						Namespace: policySnapshotNamespace,
+						Labels: map[string]string{
+							fleetv1beta1.PlacementTrackingLabel: placementName,
+						},
+					},
+				},
+				&fleetv1beta1.SchedulingPolicySnapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "other-placement-0",
+						Namespace: "other-namespace",
+						Labels: map[string]string{
+							fleetv1beta1.PlacementTrackingLabel: "other-placement",
+						},
+					},
+				},
+				// Add cluster-scoped snapshots to test mixed scenario
+				&fleetv1beta1.ClusterSchedulingPolicySnapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-placement-0",
+						Labels: map[string]string{
+							fleetv1beta1.PlacementTrackingLabel: placementName,
+						},
+					},
+				},
+				&fleetv1beta1.ClusterSchedulingPolicySnapshot{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-placement-1",
+						Labels: map[string]string{
+							fleetv1beta1.PlacementTrackingLabel: placementName,
+						},
+					},
+				},
```

Comment:
Add/remove a ClusterSchedulingPolicySnapshot to differentiate the CSPS and SPS count so that we can ensure we pick the namspaced ones not the cluster ones?